components/modal/index.js

```diff
@@ -0,0 +1,203 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+import { noop } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { Component, createPortal } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import ModalFrame from './frame';
+import ModalHeader from './header';
+import * as ariaHelper from './aria-helper';
+import './style.scss';
+
+// Used to count the number of open modals.
+let parentElement,
+	openModalCount = 0;
+
+class Modal extends Component {
+	constructor() {
+		super( ...arguments );
+
+		this.prepareDOM();
+	}
+
+	/**
+	 * Prepares the DOM for the modals to be rendered.
+	 *
+	 * Every modal is mounted in a separate div appended to a parent div
+	 * that is appended to the document body.
+	 *
+	 * The parent div will be created if it does not yet exist, and the
+	 * separate div for this specific modal will be appended to that.
+	 */
+	prepareDOM() {
+		if ( ! parentElement ) {
```

Comment:
Out of curiosity, why do we need `parentElement` ? Why not just append and remove `this.node` to / from `document.body` directly?

*Edit:* I see this is to make it easier to exclude from applying `aria-hidden`.